src/v/raft/types.cc

```diff
@@ -591,19 +592,29 @@ void heartbeat_reply::serde_read(iobuf_parser& src, const serde::header& hdr) {
 }
 
 ss::future<> append_entries_request::serde_async_write(iobuf& dst) {
-    auto mem_batches = co_await model::consume_reader_to_memory(
-      std::move(batches()), model::no_timeout);
-
-    iobuf out;
     using serde::write;
 
-    write(out, static_cast<uint32_t>(mem_batches.size()));
-    for (auto& batch : mem_batches) {
-        // intentionally using reflection here for batches which are not yet
-        // supported with serde, but also have largely solidified.
-        reflection::serialize(out, std::move(batch));
-        co_await ss::coroutine::maybe_yield();
-    }
+    class streaming_writer {
+    public:
+        ss::future<ss::stop_iteration> operator()(model::record_batch b) {
+            reflection::serialize(_out, std::move(b));
```

Comment:
In future we need to consider using `serde` in here.

Reply:
@dotnwat

Reply:
Should I make an issue to remember and tackle it in a followup?